networking/metallb/metallb-configure-services.adoc

```diff
@@ -0,0 +1,162 @@
+[id="metallb-configure-services"]
+= Configuring services to use MetalLB
+include::modules/common-attributes.adoc[]
+:context: configure-services-metallb
+
+toc::[]
+
+As a cluster administrator, when you add a service of type `LoadBalancer`, you can control how MetalLB assigns an IP address.
+
+// Request a specific IP address
+[id="request-specific-ip-address_{context}"]
+== Request a specific IP address
+
+Like some other load-balancer implementations, MetalLB accepts the `spec.loadBalancerIP` field in the service specification.
+
+If the requested IP address is within a range from any address pool, MetalLB assigns the requested IP address.
+If the requested IP address is not within any range, MetalLB reports a warning.
+
+.Example service YAML for a specific IP address
+[source,yaml]
+----
+apiVersion: v1
+kind: Service
+metadata:
+  name: <service_name>
+  annotations:
+    metallb.universe.tf/address-pool: <address_pool_name>
+spec:
+  selector:
+    <label_key>: <label_value>
+  ports:
+    - port: 8080
+      targetPort: 8080
+      protocol: TCP
+  type: LoadBalancer
+  loadBalancerIP: <ip_address>
+----
+
+If MetalLB cannot assign the requested IP address, the `EXTERNAL-IP` for the service reports `<pending>` and running `oc describe service <service_name>` includes an event like the following example.
+
+.Example event when MetalLB cannot assign a requested IP address
+[source,terminal]
+----
+  ...
+Events:
+  Type     Reason            Age    From                Message
+  ----     ------            ----   ----                -------
+  Warning  AllocationFailed  3m16s  metallb-controller  Failed to allocate IP for "default/invalid-request": "4.3.2.1" is not allowed in config
+----
+
+[id="request-ip-address-from-pool_{context}"]
+== Request an IP address from a specific pool
+
+To assign an IP address from a specific range, but you are not concerned with the specific IP address, then you can use the `metallb.universe.tf/pool-name` annotation to request an IP address from the specified address pool.
+
+.Example service YAML for an IP address from a specific pool
+[source,yaml]
+----
+apiVersion: v1
+kind: Service
+metadata:
+  name: <service_name>
+  annotations:
+    metallb.universe.tf/pool-name: <address_pool_name>
```

Comment:
This is wrong. The right annotation name is `metallb.universe.tf/metallb.universe.tf/address-pool`
```suggestion
    metallb.universe.tf/address-pool: <address_pool_name>
```

Reply:
The same applies for any other occurrence of `metallb.universe.tf/pool-name`. They should be replaced by `metallb.universe.tf/address-pool`

Reply:
Huge thanks for pointing out the error.